sunpy/map/mapbase.py

```diff
@@ -2431,25 +2431,33 @@ def _check_axes(self, axes, allow_non_wcsaxes=False):
             Axes to validate.
         allow_non_wcsaxes : bool
             If `True`, allow ``axes`` to not be a `WCSAxes`, and warn. Otherwise
-            raise an error.
+            raise an error. Support for this is deprecated, and this keyword can be removed in
+            sunpy 3.1.
+        warn_different_wcs : bool
+            If `True`, warn if the Axes WCS is different from the Map WCS. This is only used for
+            `.plot()`, and can be removed once support is added for plotting a map on a different
+            WCSAxes.
         """
         if not axes:
             axes = wcsaxes_compat.gca_wcs(self.wcs)
 
         if not wcsaxes_compat.is_wcsaxes(axes):
+            # not WCSAxes
             if allow_non_wcsaxes:
                 if not np.array_equal(self.rotation_matrix, np.identity(2)):
                     warnings.warn("The axes of this map are not aligned to the pixel grid. "
                                   "Plot axes may be incorrect.",
                                   SunpyUserWarning)
                 warnings.warn("WCSAxes not being used as the axes object for this plot. "
-                              "Plots may have unexpected behaviour. To fix this pass "
-                              "'projection=map' when creating the axes",
-                              SunpyUserWarning)
+                              "Support for this is deprecated, and will be removed in sunpy 3.1. "
+                              "To fix this pass 'projection=map' when creating the axes.",
```

Comment:
Can you reword this?  I feel like a user might literally type `projection=map` and be confused why passing the function `map` doesn't achieve the desired result.